icu4j/main/tests/core/src/com/ibm/icu/dev/test/number/NumberFormatterApiTest.java

```diff
@@ -2095,39 +2243,57 @@ public TestCase(String locale, String unitIdentifier, String expectedGender) {
                 new TestCase("fr", "minute", "feminine"),
-                new TestCase("fr", "hour", "feminine"),
-                new TestCase("fr", "day", "masculine"),
-                // grammaticalFeatures deriveCompound "per" rule:
-                new TestCase("de", "meter-per-hour", "masculine"),
-                new TestCase("af", "meter-per-hour", null),
-                // TODO(ICU-21494): determine whether list genders behave as follows,
-                // and implement proper getListGender support (covering more than just
-                // two genders):
-                // // gender rule for lists of people: de "neutral", fr "maleTaints"
-                // new TestCase("de", "day-and-hour-and-minute", "neuter"),
-                // new TestCase("de", "hour-and-minute", "feminine"),
-                // new TestCase("fr", "day-and-hour-and-minute", "masculine"),
-                // new TestCase("fr", "hour-and-minute", "feminine"),
+            new TestCase("fr", "hour", "feminine"),
+            new TestCase("fr", "day", "masculine"),
+            // grammaticalFeatures deriveCompound "per" rule takes the gender of the
+            // numerator unit:
+            new TestCase("de", "meter-per-hour", "masculine"),
+            new TestCase("fr", "meter-per-hour", "masculine"),
+            new TestCase("af", "meter-per-hour", null), // ungendered language
+            // French "times" takes gender from first value, German takes the
+            // second. Prefix and power does not have impact on gender for these
+            // languages:
+            new TestCase("de", "square-decimeter-square-second", "feminine"),
+            new TestCase("fr", "square-decimeter-square-second", "masculine"),
+            // TODO(ICU-21494): determine whether list genders behave as follows,
+            // and implement proper getListGender support (covering more than just
+            // two genders):
+            // // gender rule for lists of people: de "neutral", fr "maleTaints"
+            // new TestCase("de", "day-and-hour-and-minute", "neuter"),
+            // new TestCase("de", "hour-and-minute", "feminine"),
+            // new TestCase("fr", "day-and-hour-and-minute", "masculine"),
+            // new TestCase("fr", "hour-and-minute", "feminine"),
         };
 
         LocalizedNumberFormatter formatter;
         FormattedNumber fn;
         for (TestCase t : cases) {
             // TODO(icu-units#140): make this work for more than just UnitWidth.FULL_NAME
+        //     formatter = NumberFormatter.with()
+        //                     .unit(MeasureUnit.forIdentifier(t.unitIdentifier))
+        //                     .locale(new ULocale(t.locale));
+        //     fn = formatter.format(1.1);
+        //     assertEquals("Testing gender with default width, unit: " + t.unitIdentifier +
+        //                      ", locale: " + t.locale,
+        //                  t.expectedGender, fn.getGender());
```

Comment:
indentation

Reply:
Done.

(VSCode's doing that one wrong, so I did it manually. I have no idea how to get VSCode to format better by default - e.g. how to make it follow clang-format style...)